tests/models/constituency_parser_test.py

```diff
@@ -34,74 +36,10 @@ def test_resolve_overlap_conflicts_greedily(self):
-        # pylint: disable=bad-continuation
-        correct_tree = {
-                "label": "S",
-                "start": 0,
-                "end": 5,
-                "children": [
-                        {
-                            "label": "NP",
-                            "start": 0,
-                            "end": 2,
-                            "children": [
-                                {
-                                    "label": "D-POS",
-                                    "start": 0,
-                                    "end": 1,
-                                    "word": "the",
-                                    "is_leaf": True
-                                },
-                                {
-                                    "label": "N-POS",
-                                    "start": 1,
-                                    "end": 2,
-                                    "word": "dog",
-                                    "is_leaf": True
-                                }
-                            ]
-                        },
-                        {
-                            "label": "VP",
-                            "start": 2,
-                            "end": 5,
-                            "children": [
-                                {
-                                    "label": "V-POS",
-                                    "start": 2,
-                                    "end": 3,
-                                    "word": "chased",
-                                    "is_leaf": True
-                                },
-                                {
-                                    "label": "NP",
-                                    "start": 3,
-                                    "end": 5,
-                                    "children": [
-                                        {
-                                            "label": "D-POS",
-                                            "start": 3,
-                                            "end": 4,
-                                            "word": "the",
-                                            "is_leaf": True
-                                        },
-                                        {
-                                            "label": "N-POS",
-                                            "start": 4,
-                                            "end": 5,
-                                            "word": "cat",
-                                            "is_leaf": True
-                                        }
-                                    ]
-                                }
-                            ]
-                        }
-                    ]
-                }
-        # pylint: enable=bad-continuation
+        correct_tree = Tree.fromstring("(S (NP (D the) (N dog)) (VP (V chased) (NP (D the) (N cat))))")
```

Comment:
Now that this test is a lot more compact, it might be worth checking some more cases.  E.g., can you get into the state mentioned above?  What happens when you don't have enough spans to cover the whole sentence?